libzfs_core/_libzfs_core.py

```diff
@@ -0,0 +1,659 @@
+    return errlist.keys()
+
+
+def lzc_get_holds(snapname):
+    '''
+    Retrieve list of *user holds* on the specified snapshot.
+
+    :param bytes snapname: the name of the snapshot.
+    :return: holds on the snapshot along with their creation times
+             in seconds since the epoch
+    :rtype: dict of bytes : int
+    '''
+    holds = {}
+    with nvlist_out(holds) as nvlist:
+        ret = _lib.lzc_get_holds(snapname, nvlist)
+    xlate.lzc_get_holds_xlate_error(ret, snapname)
+    return holds
+
+
+def lzc_send(snapname, fromsnap, fd, flags = []):
+    '''
+    Generate a zfs send stream for the specified snapshot and write it to
+    the specified file descriptor.
+
+    :param bytes snapname: the name of the snapshot to send.
+    :param fromsnap: if not None the name of the starting snapshot
+                     for the incremental stream.
+    :type fromsnap: bytes or None
+    :param int fd: the file descriptor to write the send stream to.
+    :param flags: the flags that control what enhanced features can be used
+                  in the stream.
+    :type flags: list of bytes
+
+    :raises SnapshotNotFound: if either the starting snapshot is not `None` and does not exist,
+                              or if the ending snapshot does not exist.
+    :raises NameInvalid: if the name of either snapshot is invalid.
+    :raises NameTooLong: if the name of either snapshot is too long.
+    :raises SnapshotMismatch: if ``fromsnap`` is not an ancestor snapshot of ``snapname``.
+    :raises PoolsDiffer: if the snapshots belong to different pools.
+    :raises IOError: if an input / output error occurs while writing to ``fd``.
+    :raises ValueError: if the ``flags`` contain an invalid flag name.
+
+    If ``fromsnap`` is None, a full (non-incremental) stream will be sent.
+    If ``fromsnap`` is not None, it must be the full name of a snapshot or
+    bookmark to send an incremental from, e.g. :file:`{pool}/{fs}@{earlier_snap}`
+    or :file:`{pool}/{fs}#{earlier_bmark}`.
+
+    The specified snapshot or bookmark must represent an earlier point in the history
+    of ``snapname``.
+    It can be an earlier snapshot in the same filesystem or zvol as ``snapname``,
+    or it can be the origin of ``snapname``'s filesystem, or an earlier
+    snapshot in the origin, etc.
+    ``fromsnap`` must be strictly an earlier snapshot, specifying the same snapshot
+    as both ``fromsnap`` and ``snapname`` is an error.
+
+    If ``flags`` contains *"large_blocks"*, the stream is permitted
+    to contain ``DRR_WRITE`` records with ``drr_length`` > 128K, and ``DRR_OBJECT``
+    records with ``drr_blksz`` > 128K.
+
+    If ``flags`` contains *"embedded_data"*, the stream is permitted
+    to contain ``DRR_WRITE_EMBEDDED`` records with
+    ``drr_etype`` == ``BP_EMBEDDED_TYPE_DATA``,
+    which the receiving system must support (as indicated by support
+    for the *embedded_data* feature).
```

Comment:
Hmm, possibly I could link here https://github.com/illumos/illumos-gate/blob/master/usr/src/lib/libzfs_core/common/libzfs_core.c#L441 but that would not be a stable link.